tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordEntry.cs

```diff
@@ -26,16 +26,54 @@ public class RecordEntry
 
         public RequestOrResponse Response { get; set; } = new RequestOrResponse();
 
-        public string RequestUri { get; set; }
+        private string _requestUri;
+        public string RequestUri
+        {
+            get { return this._requestUri; }
+            set {
+                // If the requestUri is being modified, set the flag to true
+                if (DebugLogger.CheckLogLevel(LogLevel.Debug) && this._requestUri != value) 
+                {
+                    requestUriIsModified = true;
+                } 
+                this._requestUri = value; 
+            }
+        }
 
         public bool IsTrack1Recording { get; set; }
 
         public RequestMethod RequestMethod { get; set; }
 
         public int StatusCode { get; set; }
+        // Flag to indicate if the requestUri has been modified
+        public bool requestUriIsModified { get; set; } = false;
+
+        /// <summary>
+        /// Checks if the RecordEntry instance or any of its properties have been modified.
+        /// Primarily used to determine if the RecordEntry has been sanitized.
+        /// </summary>
+        /// <returns>True if any modification has been made, otherwise false.</returns>
+        public bool isModified()
+        {
+            return this.requestUriIsModified || this.Request.IsModified.Headers || this.Request.IsModified.Body || this.Response.IsModified.Headers || this.Response.IsModified.Body;
+        }
 
-        public static RecordEntry Deserialize(JsonElement element)
+        /// <summary>
+        /// Initializes the IsModified flags for the RecordEntry instance.
+        /// Primarily used to determine if the RecordEntry has been sanitized.
+        /// </summary>
+        public void initializeIsModifiedFlag()
         {
+            if (DebugLogger.CheckLogLevel(LogLevel.Debug))
```

Comment:
I appreciate you figuring out how to make this work in `Debug` mode.

Anywhere that you have checks for the `isModified` or not though, I think I'm ok with you pulling back from the "only in debugging" stance just for code clarity.

The reason I say that is because tracking a bool that says whether or not an object is modified won't add more than a few bits per entry. I think it's better to have clean constructor and setter code.

So for `InitializeIsModifiedFlag()`. I don't think this should exist. We should _set it_ during sanitization. 

```
                this.requestUriIsModified = false;
                this.Request.IsModified = new RequestOrResponse.RequestOrResponseIsModified();
                this.Response.IsModified = new RequestOrResponse.RequestOrResponseIsModified();
```

Make `requestUrisModified` _default_ to false. Anything that touches the entry in the `setter` should set it to `true`.

Over in `RequestOrResponse` make `IsModified = new ModificationStatus()` a default value. Can avoid this function entirely.